components/list/list-item-drag-mixin.js

```diff
@@ -33,86 +95,299 @@ export const ListItemDragMixin = superclass => class extends superclass {
 		}
 	}
 
-	moveAfter(itemKey) {
+	moveItemAfterMe(itemKey) {
 		if (itemKey === undefined) {
 			return;
 		}
-		this._dispatchDragEvent(itemKey);
+		this._moveItem(itemKey, this.key, false);
 	}
 
-	moveBefore(itemKey) {
+	moveItemBeforeMe(itemKey) {
 		if (itemKey === undefined) {
 			return;
 		}
 
-		this._dispatchDragEvent(itemKey);
+		this._moveItem(itemKey, this.key, true);
+	}
+
+	_dragAreaClick(e) {
+		this.shadowRoot.querySelector(`#${this._itemDragId}`).activateKeyboardMode();
+		e.preventDefault();
 	}
 
-	// TODO
-	_copy() {
+	_dragBottomEnter(e) {
+		e.dataTransfer.dropEffect = 'move';
+		const dropSpots = dropSpotsFactory();
+		dropSpots.setDestination(this, move.below);
+		this._inBottomArea = true;
+	}
 
+	_dragEnd(e) {
+		const dropSpot = dropSpotsBlowUp();
+		this.beingDragged = false;
+		this.tilt = false;
+		if (dropSpot.shouldDrop(e.timeStamp)) {
+			this._moveItem(dropSpot.targetKey, dropSpot.destinationKey, dropSpot.moveBeforeDestination);
+		}
 	}
 
-	_dispatchDragEvent(detail) {
+	_dragHandleActions(e) {
+		let destinationKey;
+		switch (e.detail.action) {
+			case dragActions.active:
+				this._keyboardMode = true;
+				break;
+			case dragActions.cancel:
+			case dragActions.save:
```

Comment:
This will be where the temp movement is used. Basically, in keyboard mode I want the user to be able to press escape and have things reset. So the idea is that it will send up the event saying save the temp movement or go back. It is expect that the user component handles the different states and decides if they want to use this functionality.